### PR TITLE
Feature/172 dismiss notifications swipe

### LINKED ISSUE #172: Dismiss notification functionality
Could be auto when they've been resolved, like connection request has been accepted/declined or confirmation has been view. 
Maybe a swipe option on mobile view and a dismiss option on desktop in the option menu/three dots

### PR DESCRIPTION
**Pull request by davideastmond**
Resolves https://github.com/chingu-voyages/v25-bears-team-05/issues/172
- Implemented dismiss function to delete notification
- Implemented swiping on mobile (rough)
- Implemented three-dots context menu
(note the corresponding API PR https://github.com/chingu-voyages/v25-bears-team-05-repo2/pull/56

### CODE REVIEW

File: app/src/components/notification/Notification.tsx [hunk 1]
```diff
@@ -1,28 +1,127 @@
...
+      }
+    }
+  }, [dispatch, swipingObject, props.id]);
+
+  const swipeHandlers = useSwipeable({
+    preventDefaultTouchmoveEvent: true,
+    trackMouse: true,
+    onSwipedLeft: (data: any) => {
+      if (!isSwiping && data.velocity < 1) {
+        data.event.currentTarget.style["margin-left"] = `0px`;
```
Comment (lily-law) [inline app/src/components/notification/Notification.tsx]: `data.event.currentTarget` works on touch events but is null when triggered by mouse pointer

File: app/src/components/notification/context-menu/Notification-context-menu.tsx [hunk 2]
```diff
@@ -1,8 +1,72 @@
...
+  notification,
+  onMenuOptionClicked,
+}: {
+  notification: INotificationCardData;
+  onMenuOptionClicked: (
+    notificationId: string,
+    action: NotificationContextMenuAction
+  ) => void;
+}) {
+  return (
```
Comment (lily-law) [inline app/src/components/notification/context-menu/Notification-context-menu.tsx:30]: This menu is good, nice touch with the tooltips, though I guess for mobile/touch screen users we could do with a conformation popup? Also for now if you wanted could put 'not yet implemented' in the confirm boxes

**Review by lily-law — COMMENTED** [review]
Looking pretty :cool: just a few comments

File: app/src/components/notification/Notification.tsx (hunk 1 above)
Comment (davideastmond) [reply app/src/components/notification/Notification.tsx]: ~Just looking for a bit of context for this one: Is this crashing for you? I was thinking that in mouse pointer contexts, the user would use the context menu.~

Comment (davideastmond) [reply app/src/components/notification/Notification.tsx]: Ahh ok I see what is happening now!

Comment (davideastmond) [reply app/src/components/notification/Notification.tsx]: I did a null check. Just to confirm - do we want mouse-swiping in this scenario?

Comment (lily-law) [reply app/src/components/notification/Notification.tsx]: Makes sense to not have mouse swiping here

File: app/src/components/notification/context-menu/Notification-context-menu.tsx (hunk 2 above)
Comment (davideastmond) [reply app/src/components/notification/context-menu/Notification-context-menu.tsx:30]: This is just in response to a context menu action (but not the swipe, right?)

File: app/src/components/notification/Notification.tsx [hunk 3]
```diff
@@ -1,48 +1,198 @@
...
+    }
+  }, [dispatch, swipingObject, props.id, isSwiping]);
+
+  const swipeHandlers = useSwipeable({
+    preventDefaultTouchmoveEvent: true,
+    trackMouse: true,
+    trackTouch: true,
+    onSwipedLeft: (data: any) => {
+      if (!isSwiping && data.velocity < 1) {
+        if (data.event.currentTarget) {
```
Comment (lily-law) [inline app/src/components/notification/Notification.tsx]:
```suggestion
        if (data.event?.currentTarget?.style) {
```

Comment (lily-law) [reply app/src/components/notification/Notification.tsx]: Without this check I still get `TypeError: Cannot set property 'margin-left' of undefined`
To reproduce, with Chrome's device toolbar switched off, click and drag text to left

**Review by lily-law — APPROVED** [review]
Just the one bug still, otherwise :cool:

**Pull request merged**